.class public final Landroid/support/v4/content/ad;
.super Ljava/util/concurrent/FutureTask;
.source ""


# annotations
.annotation system Ldalvik/annotation/Signature;
    value = {
        "Ljava/util/concurrent/FutureTask",
        "<TResult;>;"
    }
.end annotation


# instance fields
.field public final synthetic a:Landroid/support/v4/content/aa;


# direct methods
.method public constructor <init>(Landroid/support/v4/content/aa;Ljava/util/concurrent/Callable;)V
    .locals 0

    .prologue
    .line 13432
    iput-object p1, p0, Landroid/support/v4/content/ad;->a:Landroid/support/v4/content/aa;

    invoke-direct {p0, p2}, Ljava/util/concurrent/FutureTask;-><init>(Ljava/util/concurrent/Callable;)V

    return-void
.end method


# virtual methods
.method public final done()V
    .locals 2

    .prologue
    .line 13433
    :try_start_0
    invoke-virtual {p0}, Landroid/support/v4/content/ad;->get()Ljava/lang/Object;

    move-result-object v1

    .line 13434
    iget-object v0, p0, Landroid/support/v4/content/ad;->a:Landroid/support/v4/content/aa;

    invoke-static {v0, v1}, Landroid/support/v4/content/aa;->m$c$0(Landroid/support/v4/content/aa;Ljava/lang/Object;)V

    .line 13435
    :goto_0
    return-void
    :try_end_0
    .catch Ljava/lang/InterruptedException; {:try_start_0 .. :try_end_0} :catch_3
    .catch Ljava/util/concurrent/ExecutionException; {:try_start_0 .. :try_end_0} :catch_0
    .catch Ljava/util/concurrent/CancellationException; {:try_start_0 .. :try_end_0} :catch_1
    .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_2

    .line 13436
    :catch_0
    move-exception v0

    .line 13437
    new-instance p0, Ljava/lang/RuntimeException;

    const-string v1, "An error occurred while executing doInBackground()"

    invoke-virtual {v0}, Ljava/util/concurrent/ExecutionException;->getCause()Ljava/lang/Throwable;

    move-result-object v0

    invoke-direct {p0, v1, v0}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;Ljava/lang/Throwable;)V

    throw p0

    .line 13438
    :catch_1
    iget-object v1, p0, Landroid/support/v4/content/ad;->a:Landroid/support/v4/content/aa;

    const/4 v0, 0x0

    invoke-static {v1, v0}, Landroid/support/v4/content/aa;->m$c$0(Landroid/support/v4/content/aa;Ljava/lang/Object;)V

    goto :goto_0

    .line 13439
    :catch_2
    move-exception p0

    .line 13440
    new-instance v1, Ljava/lang/RuntimeException;

    const-string v0, "An error occurred while executing doInBackground()"

    invoke-direct {v1, v0, p0}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;Ljava/lang/Throwable;)V

    throw v1

    .line 13441
    :catch_3
    goto :goto_0
.end method
